### PR TITLE
KAFKA-9868: Reduce transaction log partitions for embed broker

### PR DESCRIPTION
**Pull request by abbccdda**
This PR tries to fix the flaky EOSUncleanShutdownIntegrationTest.shouldWorkWithUncleanShutdownWipeOutStateStore by making the bootstrapping of the test to be less painful with fewer number of partitions of txn log.

The integration test time on local also reduced one second.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/test/java/org/apache/kafka/streams/integration/utils/EmbeddedKafkaCluster.java [hunk 1]
```diff
@@ -98,6 +98,7 @@ public void start() throws IOException, InterruptedException {
         putIfAbsent(brokerConfig, KafkaConfig$.MODULE$.GroupInitialRebalanceDelayMsProp(), 0);
         putIfAbsent(brokerConfig, KafkaConfig$.MODULE$.OffsetsTopicReplicationFactorProp(), (short) 1);
         putIfAbsent(brokerConfig, KafkaConfig$.MODULE$.OffsetsTopicPartitionsProp(), 5);
+        putIfAbsent(brokerConfig, KafkaConfig$.MODULE$.TransactionsTopicPartitionsProp(), 5);
```
Comment (abbccdda) [inline streams/src/test/java/org/apache/kafka/streams/integration/utils/EmbeddedKafkaCluster.java:101]: This is the actual fix, other parts are just side cleanups.

**Comment by abbccdda** [comment]
@vvcephei @mjsax Call for a review

**Comment by mjsax** [comment]
Retest this please.

**Review by mjsax — APPROVED** [review]
LGTM. Let's see if this helps. If yes, we might want to do a similar thing for all EOS integration tests?

File: streams/src/test/java/org/apache/kafka/streams/integration/utils/EmbeddedKafkaCluster.java [hunk 2]
```diff
@@ -84,7 +84,7 @@ public EmbeddedKafkaCluster(final int numBrokers,
     /**
      * Creates and starts a Kafka cluster.
      */
-    public void start() throws IOException, InterruptedException {
+    public void start() throws IOException {
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/integration/utils/EmbeddedKafkaCluster.java:87]: If we would have written `throws Exception` from the beginning on, this change would not be necessary... (Just to back up my preferred coding stile to only use `throws Exception` in tests.)

**Comment by abbccdda** [comment]
Run 350 times on local and no failure.

File: streams/src/test/java/org/apache/kafka/streams/integration/utils/EmbeddedKafkaCluster.java (hunk 2 above)
Comment (abbccdda) [reply streams/src/test/java/org/apache/kafka/streams/integration/utils/EmbeddedKafkaCluster.java:87]: Lol, makes sense.

**Comment by abbccdda** [comment]
@mjsax I suppose this should help for all integration tests, as they share `EmbeddedBroker`

**Comment by abbccdda** [comment]
Got 2/3 green, one failed test is due to
```
kafka.api.SaslSslConsumerTest.testCoordinatorFailover

java.lang.AssertionError: expected:<None> but was:<Some(org.apache.kafka.clients.consumer.CommitFailedException: Offset commit cannot be completed since the consumer is not part of an active group for auto partition assignment; it is likely that the consumer was kicked out of the group.)>
```

**Pull request merged**